qualtran/surface_code/t_factory.py

```diff
@@ -0,0 +1,52 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import math
+
+from attrs import field, frozen
+
+from qualtran.surface_code.algorithm_summary import AlgorithmSummary
+from qualtran.surface_code.magic_state_factory import MagicStateFactory
+
+
+@frozen
+class TFactory(MagicStateFactory):
+    """A magic state factory for T states.
```

Comment:
How does this relate to the existing `CCZ2TFactory(MagicStateFactory)`? That one is a very specific construction from a particular paper. Is this supposed to be just a generic attributes-based T factory?

Reply:
The CCZ factory is as you said describies the implementation. This class is a simple data class description